src/Loader.php

```diff
@@ -192,11 +196,66 @@ public function do_not_delete_autosave( $delete = null, $post = null, $force_del
 	 */
 	public static function theme_file_ver( string $rel_path ): int {
 		$filepath = trailingslashit( get_template_directory() ) . $rel_path;
-		$ctime    = filectime( $filepath );
-		if ( $ctime ) {
-			return $ctime;
+
+		return self::get_timestamp( $filepath );
+	}
+
+	/**
+	 * Get timestamp of a file.
+	 *
+	 * @param string $path The path of the file.
+	 *
+	 * @throws RuntimeException If the file doesn't exist, or filectime fails in some other way.
+	 * @return int Timestamp of last file change.
+	 */
+	private static function get_timestamp( string $path ): int {
+		$ctime = filectime( $path );
+
+		if ( ! $ctime ) {
+			throw new RuntimeException( "Tried to get file change time of {$path} but failed to." );
```

Comment:
You could have included the runtime exception in `theme_file_ver` directly, what made you extract this part ? (I feel like the function name is more explicit about what it does, but in the context of the class `get_file_timestamp()` might be better, and in the context of `wp_enqueue_style` `theme_file_ver()` seemed more versatile if a change about versioning process was needed)

Reply:
> what made you extract this part ? 

I included the exception in that function just to avoid duplicating it, it's a private function that is used by 3 functions, for all of which it makes sense to throw the same exception.

> get_file_timestamp() might be better

The function argument already makes it clear that this acts on a file. In a lot of cases it can really help reduce redundancy if you regard the parameter as part of the function name, as it will be present both in the declaration and the usages (assuming the variable you pass to it is accurately named). Especially if the argument is a class this helps reducing repetition e.g. `get_file_timestamp(File $file)` vs `get_timestamp(File $file)`.

> and in the context of wp_enqueue_style theme_file_ver() seemed more versatile if a change about versioning process was needed

Not sure if I understand this correctly, can you elaborate?

Reply:
> The function argument already makes it clear that this acts on a file.  

Ah that's interesting, usually I'm reasoning more in term of methods for which the main subject is the class itself (like `File::get_timestamp()` for example, and the rest of the arguments are more like options), my guess is that you're displaying argument names in phpstorm ? (I don't at the moment, but I understand your point in that context)

> Not sure if I understand this correctly, can you elaborate?

I just meant that wp expects this parameter to be a file version, so naming the function for what it gives made sens to me (whatever it returns that will act as is, timestamp, version, anything else) as we wouldn't have to change its name if we want it to return a version number for example, but again, if you display arguments names I see your point.

I was saying this somewhere else, once we have PSR-4 everywhere we could extract those utilities in specific classes and reuse them in gutenberg-blocks too.

Reply:
> so naming the function for what it gives made sense to me

I did name the function for what it gives, as it gives the timestamp :D But you're right that without having the word "version" in there somewhere it's not that clear. It also doesn't help that WP abbreviates it to `$ver`... So I instead assigned  the return value to a variable.